python_modules/dagster/dagster/core/asset_defs/asset_group.py

```diff
@@ -109,8 +109,8 @@ def __init__(
         assets = with_resources(assets, resource_defs)
         source_assets = with_resources(source_assets, resource_defs)
 
-        self._assets = assets
-        self._source_assets = source_assets
+        self._assets = set(assets)
```

Comment:
Since we are making this change, we should cast the `AssetGroup.assets` and `AssetGroup.source_assets` properties to be lists.

Docstrings (and maybe examples?) reference these as lists

Reply:
yeah you're right I refactored this to not do it that way